### PR TITLE
518 native + refactored docker installer + docs

### LINKED ISSUE #518: resurrect native system gips installer for py3 + gippy 1
`install.sh` wasn't great, but it was all we had.

Resurrect that script for native gips installation (ie 'bare metal').

Start by supporting conventional installs onto ubuntu 18.04 (ie install gips into `/usr/` or wherever python stuffs things if you `setup.py install` as root).

### PR DESCRIPTION
**Pull request by ra-tolson**
Fixes #518.  ~~Presently this branch rests atop the fix for #515; the former should merge first.~~

This has been rebased after #516 merged; stand by.

### CODE REVIEW

**Comment by ra-tolson** [comment]
~~PS - I need to add this is a 'first iteration':  It solves the problem just enough to have a native installer for python 3 + gippy 1.0.  It doesn't:~~

It does do these things now:

* install sixs
* refactor the dockerfiles to call into the installer

It doesn't:

* make use of any of the advances in the dhscratch2 branch

**Comment by ra-tolson** [comment]
I have some more commits incoming; I think reaching CI pipeline passage using the new installer would be a nice checkpoint.

**Comment by ra-tolson** [comment]
See CI pipeline 90721435 for passing the test suite.  Current state:

* Breaks installation scripts into three parts (system dependencies, python dependencies, and gips installation proper); this way hopefully we (and users) can call the bits separately if they wish (next step is virtualenvs for instance).
* Thus you can take or leave the bits of system installation you need for your use case.
* since pip3 could be installed already, it checks first before assuming ubuntu package is wanted.
* intentionally broke things into short nearly logic-free scripts instead of trying to write installer logic in bash because bash is very hard to get right.

**Comment by ra-tolson** [comment]
That commit describes how to use the install scripts & how to do up a virtualenv for users that want that.  I also ran the full test suite locally (everything but fetch & modisndti) and it all worked.  Also I ran a modis fetch to confirm earthdata etc continues to function.

File: setup.py [hunk 1]
```diff
@@ -74,13 +72,14 @@
...
-        'gippy',
+        'gippy>=1.0',
         'homura==0.1.3',
         'python-dateutil',
         'pydap==3.2',
         'pysolar==0.6',
         'dbfread==2.0.7',
-        # TODO this format doesn't seem to work and I can find no documentation for it:
+        # this format doesn't work with the old pip3 included with ubuntu; to fix it, probably
+        # don't install ubuntu's pip3 and instead do: https://pip.pypa.io/en/stable/installing/
```
Comment (ircwaves) [inline setup.py:82]: I believe this is the way to go.  Not required for merge though.

**Review by ircwaves — APPROVED** [review]

**Pull request merged**